pallets/elections/src/traits.rs

```diff
@@ -0,0 +1,76 @@
+use frame_support::{pallet_prelude::Get, traits::Currency};
+use sp_std::vec::Vec;
+
+pub type SessionId = u32;
+pub type EraId = u32;
+
+pub trait SessionInfoProvider<T: frame_system::Config> {
+    fn current_session() -> SessionId;
+    fn current_committee() -> Vec<T::AccountId>;
+}
+
+impl<T> SessionInfoProvider<T> for pallet_session::Pallet<T>
+where
+    T: pallet_session::Config,
+    T::ValidatorId: Into<T::AccountId>,
+{
+    fn current_session() -> SessionId {
+        pallet_session::Pallet::<T>::current_index()
+    }
+
+    fn current_committee() -> Vec<T::AccountId> {
+        pallet_session::Validators::<T>::get()
+            .into_iter()
+            .map(|a| a.into())
+            .collect()
+    }
+}
+
+pub trait ValidatorRewardsHandler<T: frame_system::Config> {
+    fn all_era_validators(era: EraId) -> Vec<T::AccountId>;
```

Comment:
Shall we comment here that order of validators is not upfront determined, yet deterministic for this same runtime?

Reply:
Right, it is more like public knowledge.